pkg/config/map.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+// KeySuite is a named key suite with file paths.
+type KeySuite struct {
+	Name            string `json:"name"`
+	KeyPath         string `json:"keyPath"`
+	CertificatePath string `json:"certPath"`
+}
+
+// KeyMap is a set of KeySuite indexed by name.
+// The overall performance is O(n) while the order of entries is persevered.
+type KeyMap []KeySuite
+
+// Append appends a uniquely named KeySuite to the map.
+// Return true if new values are appended.
+func (m *KeyMap) Append(name, keyPath, certPath string) bool {
+	for _, ref := range *m {
+		if ref.Name == name {
+			return false
+		}
+	}
+	*m = append(*m, KeySuite{
+		Name:            name,
+		KeyPath:         keyPath,
+		CertificatePath: certPath,
+	})
+	return true
+}
+
+// Remove removes a named path from the map.
+// Return true if an entry is found and removed.
+func (m *KeyMap) Remove(name string) bool {
+	for i, ref := range *m {
+		if ref.Name == name {
+			*m = append((*m)[:i], (*m)[i+1:]...)
+			return true
+		}
+	}
+	return false
+}
+
+// Get return the paths of the given name.
+// Return true if found.
+func (m KeyMap) Get(name string) (string, string, bool) {
+	for _, ref := range m {
+		if ref.Name == name {
+			return ref.KeyPath, ref.CertificatePath, true
+		}
+	}
+	return "", "", false
+}
+
+// CertificateReference is a named file path.
+type CertificateReference struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
+
+// CertificateMap is a set of CertificateReference indexed by name.
+// The overall performance is O(n) while the order of entries is persevered.
+type CertificateMap []CertificateReference
```

Comment:
Why do we need both CertificateMap and KeyMap? Can we reuse KeyMap to store certificates with key as nil.

Reply:
The JSON property names are different for `CertificateMap` and `KeyMap`. Although we can use `interface{}` and have a generic struct, `Marshal` works but `Unmarshal` simply does not work.